src/reference/asciidoc/amqp.adoc

```diff
@@ -361,6 +361,19 @@ The `ConnectionFactory` argument can be used to distinguish target connection na
 By default a `beanName` of the `AbstractConnectionFactory` and an internal counter are used to generate `connection_name`.
 The `<rabbit:connection-factory>` namespace component is also supplied with the `connection-name-strategy` attribute.
 
+
+The connection might be blocked for interaction from the Broker according to the https://www.rabbitmq.com/memory.html[Memory Alarm].
+Starting with _version 2.0_, the `org.springframework.amqp.rabbit.connection.Connection` can be supplied with the `com.rabbitmq.client.BlockedListener` s to to be notified for connection block and unblock events.
+In addition the `AbstractConnectionFactory` emits a `ConnectionBlockedEvent` and `ConnectionUnblockedEvent`, respectively, via its internal `BlockedListener` implementation.
+These can be useful in application logic to react appropriately to the problems on the Broker and perform some maintenance actions for example.
```

Comment:
~~These can be useful in application logic to react appropriately to the problems on the Broker and perform some maintenance actions for example.~~
These allow you to provide application logic to react appropriately to problems on the broker and take some corrective actions for example.